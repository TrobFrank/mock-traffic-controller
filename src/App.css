:root {
  --green: rgb(41, 252, 41);
  --yellow: rgb(252, 248, 41);
  --red: rgb(230, 26, 26);
  --black: #333;
}

#root {
  max-width: 1280px;
  margin: 0 auto;
  padding: 2rem;
  text-align: center;
}

body {
  font-size:24px;
}

.intersection {
  position:relative;
  width:10em;
  height:10em;
  border:1px solid var(--black);
}

.street {
  position: absolute;
  background:var(--black);
  width: 1em;
  top: 0;
  left: 50%;
  height: 100%;
}

.trafficLight:after {
  content: '';
  display: block;
  width: 1em;
  height: 1em;
  border-radius: 1em;
}

.trafficLight.green:after  {
  background:var(--green);
}

.trafficLight.yellow:after {
  background: var(--yellow);
}

.trafficLight.red:after {
  background: var(--red);
}


.street.ns {
  transform: translate(-50%, 0)
}

.street.ew {
  height:1em;
  width:100%;
  left:0;
  top:50%;
  transform: translate(0%, -50%)
}
.street.ns .streetName {
  position: absolute;
  top:-2em;
  white-space: nowrap;
}

.street.ew .streetName {
  position:absolute;
  right:-5em;
  white-space: nowrap;
}