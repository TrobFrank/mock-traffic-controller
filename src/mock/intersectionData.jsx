const intersectionData = [
    {
        intersection: [
            {
                id: 1,
                label: "Ocean Avenue",
                direction: "ns",
                lanes: 2,
                crossWalk: false,
                leftArrow: false,
            },
            {
                id: 2,
                label: "Boulevard of Broken Dreams",
                direction: "ew",
                lanes: 2,
                crossWalk: false,
                leftArrow: false,
            }
        ]
    },   
    {
        intersection: [
            {
                id: 3,
                label: "Electric Avenue",
                direction: "ns",
                lanes: 2,
                crossWalk: false,
                leftArrow: true,
            },
            {
                id: 4,
                label: "Dream Street",
                direction: "ew",
                lanes: 2,
                crossWalk: false,
                leftArrow: true,
            }
        ]
    },
]

export default intersectionData