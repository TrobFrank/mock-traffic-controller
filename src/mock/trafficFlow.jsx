const trafficFlow = [
    {
      phase: 0, 
      ns: { color:'green'},
      ew: { color:'red'},
      duration: 6000
    },
    {
      phase: 1, 
      ns: { color:'yellow'},
      ew: { color:'red'},
      duration: 4000
    }, 
    {
      phase: 2, 
      ns: { color:'red'},
      ew: { color:'red'},
      duration: 2000
    }, 
    {
      phase: 3, 
      ns: { color:'red'},
      ew: { color:'green'},
      duration: 6000
    },     
    {
      phase: 4, 
      ns: { color:'red'},
      ew: { color:'yellow'},
      duration: 4000
    },  
    {
      phase: 5, 
      ns: { color:'red'},
      ew: { color:'red'},
      duration: 1000
    },          
  ]

export default trafficFlow;