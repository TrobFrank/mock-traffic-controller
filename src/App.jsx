import TrafficController from './components/TrafficController';
import './App.css'

function App() {
  return <TrafficController />
}
export default App
