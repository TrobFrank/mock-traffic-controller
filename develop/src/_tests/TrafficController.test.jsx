import { describe, expect, it } from 'vitest'
import App from '../App';
import intersectionData from '../mock/intersectionData'
import trafficFlow from '../mock/trafficFlow'
import { render, screen } from '@testing-library/react';

describe('traffic controller', () => {
  it('trafficFlow defined as array', () => {
      expect(Array.isArray(trafficFlow)).toBe(true)
  })

  it('intersectionData defined as array', () => {
    expect(Array.isArray(intersectionData)).toBe(true)
  }) 

  it('uses flex on intersections div', async () => {
    render(<App />)
    const element = screen.getByRole('main')
    expect(element.className).toEqual('intersections')
    expect(getComputedStyle(element).display).toEqual('flex')
  })

})