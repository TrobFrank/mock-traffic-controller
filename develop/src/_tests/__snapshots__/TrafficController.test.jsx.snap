// Vitest Snapshot v1

exports[`traffic controller > snapshot 1`] = `
{
  "foo": "bar",
}
`;
